conans/tools.py

```diff
@@ -183,15 +183,27 @@ def vcvars_command(settings):
                                  % (existing_version, compiler_version))
     else:
         env_var = "vs%s0comntools" % compiler_version
+
+        vs_path = None
+        if env_var == 'vs150comntools' and not env_var in os.environ:
+            try:
```

Comment:
Python is "magic" :) You can just add some "_cached_path" to the function itself. I am talking about a runtime cached value, not something in the system. So it will execute once for every conan run, not once per installation, so no concerns about re-installations. Something like:

```python
def some_method():
    if not hasattr(some_method, "_cached_return_value"):
        print "Expensive operation"
        some_method._cached_return_value = 42

    return some_method._cached_return_value


print some_method()
print some_method()
print some_method()
```